crates/trigger/src/runtime_config/sqlite.rs

```diff
@@ -20,30 +18,41 @@ pub(crate) fn build_component(
         .into_iter()
         .collect();
     execute_statements(sqlite_statements, &databases)?;
-    Ok(SqliteComponent::new(databases))
+    let connections_store =
+        Arc::new(SimpleConnectionsStore(databases)) as Arc<dyn ConnectionsStore>;
+    Ok(SqliteComponent::new(move |_| connections_store.clone()))
+}
+
+/// A `ConnectionStore` based on a `HashMap`
+struct SimpleConnectionsStore(HashMap<String, Arc<dyn Connection>>);
+
+impl ConnectionsStore for SimpleConnectionsStore {
+    fn get_connection(&self, database: &str) -> Option<Arc<(dyn Connection + 'static)>> {
+        self.0.get(database).cloned()
+    }
 }
 
 fn execute_statements(
     statements: &[String],
-    databases: &HashMap<String, Arc<dyn spin_sqlite::ConnectionManager>>,
+    databases: &HashMap<String, Arc<dyn spin_sqlite::Connection>>,
 ) -> anyhow::Result<()> {
-    if !statements.is_empty() {
-        if let Some(default) = databases.get("default") {
-            let c = default.get_connection().context(
-                "could not get connection to default database in order to execute statements",
-            )?;
-            for m in statements {
-                if let Some(file) = m.strip_prefix('@') {
-                    let sql = std::fs::read_to_string(file).with_context(|| {
-                        format!("could not read file '{file}' containing sql statements")
-                    })?;
-                    c.execute_batch(&sql)
-                        .with_context(|| format!("failed to execute sql from file '{file}'"))?;
-                } else {
-                    c.query(m, Vec::new())
-                        .with_context(|| format!("failed to execute statement: '{m}'"))?;
-                }
-            }
+    if statements.is_empty() {
+        return Ok(());
+    }
+    let Some(default) = databases.get("default") else { return Ok(()) };
```

Comment:
Why do we return `Ok(())` if there is no default database here?

Reply:
I guess we should probably be emitting a warning or an error. This PR doesn't change this behaviour though so I think we should tackle this in a follow up.